rpm/gvisor-tap-vsock.spec

```diff
@@ -92,8 +92,8 @@ CGO_CFLAGS=$(echo $CGO_CFLAGS | sed 's/-specs=\/usr\/lib\/rpm\/redhat\/redhat-an
 export CGO_CFLAGS+=" -m64 -mtune=generic -fcf-protection=full"
 %endif
 
-# reset LDFLAGS for plugins and gvisor binaries, but ensure gvproxy --version and such is set
-LDFLAGS="-X github.com/containers/gvisor-tap-vsock/pkg/types.PackageVersion=%{version}"
+# reset LDFLAGS for plugins and gvisor binaries
+LDFLAGS=''
```

Comment:
We removed this because we are sure that the tarball source for rpm would have `gitArchiveVersion` set so as part of spec we don't have to provide anything?

Reply:
Another thing which is not clear when removing this is assume that I want to create rpm package for v1.5 so I tag it and then it `git describe` should show `v1.5` ?

Reply:
> We removed this because we are sure that the tarball source for rpm would have gitArchiveVersion set so as part of spec we don't have to provide anything?

Yes, that's correct, the tarball is assumed to have `gitArchiveVersion` set. If this is not always true, we can revert back to setting `-X github.com/...`

> Another thing which is not clear when removing this is assume that I want to create rpm package for v1.5 so I tag it and then it git describe should show v1.5 ?

```
$ git tag -s v0.8.0
$ git describe HEAD
v0.8.0
```

Then if I push this to github, and generate a release, `version.go` will contain `gitArchiveVersion="v0.8.0"` instead of `gitArchiveVersion = "$Format:%(describe)$"`

Reply:
Looks like when I tried it for `crc-org/crc` I am not getting expected result. I checked out `v2.30.0` tag and describe shows me something unexpected.

```
 $ git checkout v2.30.0 
Note: switching to 'v2.30.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at b6532a cut v2.30.0

$ git describe HEAD
v1.23.0-1666-gb6532a
```

By going through `man git-describe` looks like `git describe only shows annotated tags. For more information about creating annotated tags see the -a and -s options to git-tag(1)` which means we have to create annotated tags otherwise it will fail. Might be mention somewhere so that if someone else from the team creating the tag know that it should be annotated one.

Reply:
> Might be mention somewhere so that if someone else from the team creating the tag know that it should be annotated one.

Even better to create GPG signed tags